temoa/extensions/economies_of_scale/core/data_puller.py

```
@@ -0,0 +1,181 @@
+)
+
+if TYPE_CHECKING:
+    from temoa.core.model import TemoaModel
+    from temoa.extensions.economies_of_scale.core.model import EOSModel
+    from temoa.types import Period, Region, Technology, Vintage
+
+logger = logging.getLogger(__name__)
+
+
+# --- Poll cost results ---
+def poll_costs(
```

Ruff (0.15.21)

[warning] Too many branches (15 > 12)

(PLR0912)